constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/Violation.java

```diff
@@ -1,28 +1,36 @@
 package gov.nasa.jpl.aerie.constraints.model;
 
+import gov.nasa.jpl.aerie.constraints.time.Interval;
 import gov.nasa.jpl.aerie.constraints.time.Windows;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.StreamSupport;
 
 public final class Violation {
   public final List<Long> activityInstanceIds;
   public final List<String> resourceNames;
-  public final Windows violationWindows;
+  public final List<Interval> violationWindows;
 
   public Violation(final List<Long> activityInstanceIds, final List<String> resourceNames, final Windows violationWindows) {
     this.activityInstanceIds = new ArrayList<>(activityInstanceIds);
     this.resourceNames = new ArrayList<>(resourceNames);
-    this.violationWindows = violationWindows;
+    this.violationWindows = StreamSupport.stream(violationWindows.spliterateEqualTo(true), false).toList();
```

Comment:
I don't know the difference between a spliterator and an iterator. I do know that neither Iterators or Spliterators have any kind of `toList` or `collect` method though, so in order to collect we have to convert to a Stream. An `StreamSupport.stream(...)` only accepts Spliterators, whatever they are.

Reply:
It looks like `Spliterators` are used if you want to split the dataset up so you can do multithreaded processing. As for collecting it to a list, was there a reason you chose to use a stream instead of iterating through the iterator and adding each element to the list?

Reply:
Streams can in principle be processed either sequentially or in parallel. The Spliterator interface allows a spliterator covering some range of data to be logically split into two spliterators covering their own independent subranges. Because Streams have this capability, if you want to use the Stream APIs, you need Spliterators -- even if your own data structure doesn't support that kind of range splitting.

Reply:
I'm just used to the functional style of filtering, mapping, and collecting iterators. If Java wasn't 26 years old I'd write `violationWindows.iterateEqualTo(true).collect<List>()`. I suppose the imperative way is probably better here.